pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -113,6 +112,11 @@ func (r *InfraClusterController) reconcile(ctx context.Context, log logr.Logger)
 		return ctrl.Result{}, nil
 	}
 
+	if errors.Is(err, errPlatformNotSupported) {
```

Comment:
We will never get here, because of line 104, you'll need to make some adjustments to the logic.

Looking at the content of ensueInfraCluster, it looks like it already returns an unsupported platform error of some description, but that's not being handled here either, should they use the same error?

Whn would `infraCluster` be nil on line 108?